js/countdown.js

```diff
@@ -2,6 +2,22 @@ document.addEventListener("DOMContentLoaded", async () => {
   const configResponse = await fetch("./config.json");
   const config = await configResponse.json();
 
+  // this is used by our automated end-to-end tests to control the "current time" in a convenient fashion
+  //
+  // NOTE: the backend checks the current time independently when submitting registrations, so this is perfectly safe
+  //
+  if (window.location.search.indexOf('currentTime') !== -1) {
+    window['mockCurrentTime'] = '' + window.location.search;
+  }
+  if (window['mockCurrentTime']) {
+    config.timeServer += '' + window['mockCurrentTime'];
+
+    const elements = document.querySelectorAll(".add-current-time-mock-if-set");
+    for (let i = 0; i < elements.length; i++) {
+      elements[i].href += '' + window['mockCurrentTime'];
+    }
+  }
```

Comment:
Does any external script require the `mockCurrentTime` being set on the window object? If not, this can be written more concisely like this:

```javascript
  if (window.location.search.indexOf('currentTime') !== -1) {
    config.timeServer += window.location.search;

    const elements = document.querySelectorAll(".add-current-time-mock-if-set");
    for (let i = 0; i < elements.length; i++) {
      elements[i].href += window.location.search;
    }
  }
```

Reply:
Good point. I initially hoped to only need to pass it to the first page and store it in the window object for subsequent pages, but that ended up not working for the ui tests because at least for firefox it doesn't survive loading a new page. I'll fix that.